lib/pundit/authorization.rb

```diff
@@ -70,9 +70,10 @@ def authorize(record, query = nil, policy_class: nil)
     # Allow this action not to perform authorization.
     #
     # @see https://github.com/varvet/pundit#ensuring-policies-and-scopes-are-used
+    # @param reason [Symbol, String] optional reason why the authorization was skipped
     # @return [void]
-    def skip_authorization
-      @_pundit_policy_authorized = :skipped
+    def skip_authorization(reason = :skipped)
+      @_pundit_policy_authorized = reason
```

Comment:
The prefixing of this variable with an underscore is a convention which means "This is not public API, please do not rely on this variable". If we were to accept this change (and I'm not saying that it's something we're prepared to maintain) then this underscore should be removed first.

Reply:
We would be happy to introduce `attr_accessors` for these fields if you think that's acceptable?

Reply:
https://github.com/varvet/pundit/pull/726